components/public-api/README.md

```diff
@@ -0,0 +1,101 @@
+# Public-API
+
+# Example Flows
+This section gives examples how clients would use this API.
+
+## Create Workspace
+
+### Not prebuild aware
+```go
+workspaces.CreateWorkspace({
+    idempotency_token: "<random_string>",
+    context_url: "https://github.com/gitpod-io/gitpod",
+})
+```
+
+### Prebuild aware but no log support
+```go
+workspaces.CreateWorkspace({
+    idempotency_token: "<random_string>",
+    context_url: "https://github.com/gitpod-io/gitpod",
+    prebuild: {
+        if_available: true,
+    }
+})
+```
+
+### Prebuild aware with log support
+```go
+contextURL := "https://github.com/gitpod-io/gitpod"
+prb := prebuilds.GetRunningPrebuild({ context_url: contextURL })
+logs := prebuilds.ListenToPrebuild({ context_url: contextURL, prebuild_id: prb.PrebuildID })
```

Comment:
Is `context_url` a leftover or why would I provide it in addition to the prebuild id?